pkg/ccl/changefeedccl/checkpoint/checkpoint.go

```diff
@@ -78,11 +78,11 @@ type SpanForwarder interface {
 func Restore(sf SpanForwarder, checkpoint *jobspb.TimestampSpansMap) error {
 	for ts, spans := range checkpoint.All() {
 		if ts.IsEmpty() {
-			return errors.New("checkpoint timestamp is empty")
+			return errors.AssertionFailedf("checkpoint timestamp is empty")
```

Comment:
Won't the `AssertionFailedf` cause the node to panic? Is this the behavior we want, or is there a graceful way to recover?

Reply:
It will, but this would only happen as the result of developer error. It should've been marked as an assertion error when it was first added.